packages/wasm/rs/tests/test_encode.rs

```diff
@@ -0,0 +1,244 @@
+      "Yp52qvoDPufUebLksFl7astBNEnjPVUX2e3O9O6VKeUpB0iiHQXfzOOjTEK6Xy6ks4zAG2M6jCL01flIJlxplRXCV7 sadsadsadsadasdasaaaaa"))
+      .unwrap();
+  assert_eq!(
+    [ 217, 255, 84, 104, 105, 115, 32, 105, 115, 32, 97, 32, 115, 116, 114, 32, 56, 32, 115, 116, 114,
+      105, 110, 103, 32, 111, 102, 32, 50, 53, 53, 32, 98, 121, 116, 101, 115, 32, 65, 67, 53, 51, 76,
+      103, 120, 76, 76, 79, 75, 109, 48, 104, 102, 115, 80, 97, 49, 86, 48, 110, 102, 77, 106, 88, 116,
+      110, 109, 107, 69, 116, 116, 114, 117, 67, 80, 106, 99, 53, 49, 100, 116, 69, 77, 76, 82, 74, 73,
+      69, 117, 49, 89, 111, 82, 71, 100, 57, 111, 88, 110, 77, 52, 67, 120, 99, 73, 105, 84, 99, 57, 86,
+      50, 68, 110, 65, 105, 100, 90, 122, 50, 50, 102, 111, 73, 122, 99, 51, 107, 113, 72, 66, 111, 88,
+      103, 89, 115, 107, 101, 118, 102, 111, 74, 53, 82, 75, 89, 112, 53, 50, 113, 118, 111, 68, 80, 117,
+      102, 85, 101, 98, 76, 107, 115, 70, 108, 55, 97, 115, 116, 66, 78, 69, 110, 106, 80, 86, 85, 88, 50,
+      101, 51, 79, 57, 79, 54, 86, 75, 101, 85, 112, 66, 48, 105, 105, 72, 81, 88, 102, 122, 79, 79, 106,
+      84, 69, 75, 54, 88, 121, 54, 107, 115, 52, 122, 65, 71, 50, 77, 54, 106, 67, 76, 48, 49, 102, 108,
+      73, 74, 108, 120, 112, 108, 82, 88, 67, 86, 55, 32, 115, 97, 100, 115, 97, 100, 115, 97, 100, 115,
+      97, 100, 97, 115, 100, 97, 115, 97, 97, 97, 97, 97 ],
+      writer.get_buffer().as_slice()
+  );
+}
+
+#[test]
+fn test_write_nil() {
+    let mut writer = WriteEncoder::new(&[], Context::new());
+    writer.write_nil().unwrap();
+    assert_eq!([192], writer.get_buffer().as_slice());
+}
+
+#[test]
+fn test_write_bool_true() {
+    let mut writer = WriteEncoder::new(&[0x00], Context::new());
+    writer.write_bool(&true).unwrap();
+    assert_eq!([195], writer.get_buffer().as_slice());
+}
+
+#[test]
+fn test_write_bool_false() {
+    let mut writer = WriteEncoder::new(&[], Context::new());
+    writer.write_bool(&false).unwrap();
+    assert_eq!([194], writer.get_buffer().as_slice());
+}
+
+#[test]
+fn test_write_u8() {
+    let mut writer = WriteEncoder::new(&[], Context::new());
+    writer.write_u8(&255).unwrap();
+    assert_eq!([204, 255], writer.get_buffer().as_slice());
+}
+
+#[test]
+fn test_write_u16() {
+    let mut writer = WriteEncoder::new(&[], Context::new());
+    writer.write_u16(&65535).unwrap();
+    assert_eq!([205, 255, 255], writer.get_buffer().as_slice());
+}
+
+#[test]
+fn test_write_u32() {
+    let mut writer = WriteEncoder::new(&[], Context::new());
+    writer.write_u32(&4294967295).unwrap();
+    assert_eq!([206, 255, 255, 255, 255], writer.get_buffer().as_slice());
+}
+
+#[test]
+fn test_write_i8() {
+    let mut writer = WriteEncoder::new(&[], Context::new());
+    writer.write_i8(&(-1)).unwrap();
+    assert_eq!([255], writer.get_buffer().as_slice());
+}
+
+#[test]
+fn test_write_i16() {
+    let mut writer = WriteEncoder::new(&[], Context::new());
+    writer.write_i16(&(-123)).unwrap();
```

Comment:
`-123` is not `i16`. It is still `i8`. You need to use something less than -128

Reply:
Great catch, thank you!